ToolBox/TickAggregatingConsolidator.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System.Collections.Generic;
+using QuantConnect.Data;
+using QuantConnect.Data.Consolidators;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.ToolBox
+{
+    /// <summary>
+    /// Class that uses consolidators to aggregate tick data data
+    /// </summary>
+    public abstract class TickAggregatingConsolidator
+    {
+        /// <summary>
+        /// The consolidator used to aggregate data from
+        /// higher resolutions to data in lower resolutions
+        /// </summary>
+        public abstract IDataConsolidator Consolidator { get; }
+
+        /// <summary>
+        /// The consolidated data
+        /// </summary>
+        public List<BaseData> Consolidated { get; protected set; }
+
+        /// <summary>
+        /// The resolution that the data is being aggregated into
+        /// </summary>
+        public Resolution Resolution { get; protected set; }
```

Comment:
Consider making this `private set` and having it set via the constructor of this class. Same for `Consolidated`. Also, `Consolidated` might be better typed as `IReadOnlyList` or `IEnumerable` publicly to prevent the *possibility* of external code being able to modify our data storage.